.github/workflows/release.yaml

```diff
@@ -8,14 +8,22 @@ on:
 jobs:
 
   publish-image:
+    permissions:
+      contents: read
+      packages: write # Used to push images to `ghcr.io`
+      id-token: write # Needed to create an OIDC token for keyless signing
```

Comment:
Wow. Somehow I hadn't realized you could do this. Now I feel like a sucker for adding permissions to the PAT the old fashioned way. 👍

Reply:
By default GITHUB_TOKEN permissions are very permissive.  Here we are explicitly setting permissions at a job level.  I should do a follow up PR and limit the scope of permissions to all jobs and set no permissions as default for each workflow. 

![Screenshot 2023-08-15 at 10 44 03 AM](https://github.com/akuity/kargo/assets/76892343/88fb2343-8e70-4d75-9aa8-845d9ed46705)